code/DDSCodeTester.cpp

```diff
@@ -7785,6 +7793,223 @@ void pubsub_api_example_create_entities()
+        ReturnCode_t ret;
+        ret = participant->register_service_type(service_type_support, "ServiceType");
+        if (RETCODE_OK != ret)
+        {
+            // Error
+            return;
+        }
+
+        // Create a new Service instance
+        Service* service = participant->create_service("Service", "ServiceType");
+        if (!service)
+        {
+            // Error
+            return;
+        }
+
+        // ... Create Requesters and Repliers here
+
+        // Delete the created Service
+        ret = participant->delete_service(service);
+        if (RETCODE_OK != ret)
+        {
+            // Error
+            return;
+        }
+        //!--
+    }
+
+    {
+        DomainParticipant* participant =
+            DomainParticipantFactory::get_instance()->create_participant(0, PARTICIPANT_QOS_DEFAULT);
+        TypeSupport request_type_support = TypeSupport(new CustomDataType());
+        TypeSupport reply_type_support = TypeSupport(new CustomDataType());
+        ServiceTypeSupport service_type_support(request_type_support, reply_type_support);
+        participant->register_service_type(service_type_support, "ServiceType");
+        participant->create_service("Service", "ServiceType");
+
+        //!--RPC_REQUESTER_EXAMPLE
+        ReturnCode_t ret;
+
+        // Get a valid service
+        Service* service = participant->find_service("Service");
+        if (!service)
+        {
+            // Error
+            return;
+        }
+
+        /* Create a new Requester instance */
+        RequesterQos requester_qos;
+
+        Requester* requester = participant->create_service_requester(service, requester_qos);
+        if (!requester)
+        {
+            // Error
+            return;
+        }
+
+        /* Send a new Request sample and check if a received Reply sample is a match */
+        // Make sure that all RPC Entities are enabled
+        if (!service->is_enabled())
+        {
+            ret = service->enable();
+            if (RETCODE_OK != ret)
+            {
+                // Error
+                return;
+            }
+        }
+
+        //  Enabling the service enables the registered Requester unless
+        //  the creation of the internal DataWriter and DataReader failed.
+        //  We can make sure that they have been created correctly by checking if the Requester is enabled.
+        if (!requester->is_enabled())
+        {
+            // Error
+            return;
+        }
+
+        RequestInfo expected_request_info;
+        RequestInfo received_request_info;
+        // Create a new Request sample
+        void* request_data = request_type_support->create_data();
+        ret = requester->send_request(request_data, expected_request_info);
+        if (RETCODE_OK != ret)
+        {
+            // Error
+            return;
+        }
+ 
+        // Wait until a Reply sample is received
+        requester->get_requester_reader()->wait_for_unread_message();
```

Comment:
This will fail to build unless we add a duration argument:
```suggestion
        // Wait for some time until a Reply sample is received
        requester->get_requester_reader()->wait_for_unread_message(Duration_t{3, 0});
```

Reply:
Changes applied on b894d65e1dbf09a3d72cc9a40f3d8687dcc58be6